packages/query-core/src/types.ts

```diff
@@ -46,10 +46,20 @@ export declare const dataTagSymbol: unique symbol
 export declare const dataTagErrorSymbol: unique symbol
 export declare const unsetMarker: unique symbol
 export type UnsetMarker = typeof unsetMarker
-export type DataTag<TType, TValue, TError = UnsetMarker> = TType & {
-  [dataTagSymbol]: TValue
-  [dataTagErrorSymbol]: TError
+export type AnyDataTag = {
+  [dataTagSymbol]: any
+  [dataTagErrorSymbol]: any
 }
+export type DataTag<
+  TType,
+  TValue,
+  TError = UnsetMarker,
+> = TType extends AnyDataTag
+  ? TType
+  : TType & {
+      [dataTagSymbol]: TValue
+      [dataTagErrorSymbol]: TError
+    }
```

Comment:
Fixes this: 

<img width="300" alt="image" src="https://github.com/user-attachments/assets/b39cb713-6334-4087-90b9-262dbdb5a3aa">

becomes this:

<img width="297" alt="image" src="https://github.com/user-attachments/assets/e4b107b5-e0b9-41fa-b226-09b9ab53db77">